### PR TITLE
(Updated on 2021.12.29) Many changes Can't Title Summarize

### PR DESCRIPTION
**Pull request by SoraHjort**
tl;dr: many changes were made, please review changes and test stuff out before going with a merge. 

(Due to a woops I made at one point, it's being pulled from a new sub-branch. The actual full change history log: https://github.com/SoraHjort/webcomic_reader/commits/Fixes-and-Additions/webcomic_reader.user.js )

This is about a year in the making (on and off a lot). It shouldn't have taken that long, but I kept getting side tracked, fixing comics I was reading at the time, forgetting about it, and so on.

This started off as a branch to just add and fix comics before putting in pull requests. It then went into adding several features and other stuff. I am stuck in a "what do I do next" slump, which is why I'm putting it out now. I think I'm mostly done. But I would like feedback on the changes and the like.

If it comes down to it, and the changes I've made are not acceptable, I could separate the additions/fixes for sites.

Highlights: 
1. Styling to prevent sites from screwing things up (like with the settings window)
2. Fullscreen Button
3. (Mobile) Disable/Enable this Script button at the end of the page (blame greasemonkey on mobile for the idea)
4. `@includes` have been updated to be "http*://" instead of http and https to
 a. save space when it comes to duplicates
 b. future proof in case a domain goes from http to https
 c. more strict capture than simply going "*://" to prevent running on, say, ftp:// or any other protocol.
5. primarily a mobile issue, but script no longer run when viewing just a image files.

Please give this a good look over and test it out. I'm the kind of person to be paranoid over my own work when it's this big of a change. I worry I might have broken something, despite not running into any issues in the months I've been using my dev branch exclusively). Heck I'm still paranoid if I'm doing pulls correctly.

Thankfully, I kept notes on note worthy changes( IE: not every time I had to edit because I goofed)
The full change summary: 

(Note on Site fixes and added sites, if they've changed since I've made those changes, and no longer work, sorry for the inconvenience.)

 
Sites Fixed: 
- Homestuck (Hopefully)
- Bog and George (Multi Image and commentary having images issue)
- Girl Genius (Multi-image should be working just fine now!)
- Vicki Fox
- Free Fall  (Broken on the first 300-ish pages)
- Dominic Deegan (Partially for mobile users)
- Dilbert
- Lackadaisy Cats
- Kevin and Kell (could use some styling fixes on mobile when not minimalistic)
- Mangaseeonline (accidentally removed when it just needed a http* fix)
- DMFA
- LFG
- GrrlPower
- Keychain of Creation
- Three Panel Soul (Double Check later)
- TwoKinds
- Project Future (back would got ot the start of chapters instead of the previous page)
- FreeFall (minor, some pages had two comic images)
- Devil's Panties (Multi-image pages)
- VGCats (It's back, new site layout, new comic page)
- Misfile
- Furaffinity now works on both www and non www urls

Sites Added: 
- Yosh Saga
- Artificial Incident
- Kemono Cafe
- Comic Host Xepher
- Skin Horse
- Narbonic
- Sailor Sun
- Jeanie Bottle
- Dreamland Chronicles
- Snafu Comics (main issues are built into the archive, like pages being out of order, nothing we can do)
- Alice and the Nightmare
- Not a Villian
- Mac Hall
- e621.net
- Alien Dice
- Most if not all the other Shivae comics as well
- Xepher comics in general

Site tweaks:
- Furaffinity tries to default to comic navigation options. If it fails to find a next/back comic page, it'll default to gallery viewing mode
- DHS Comics @includes have been condensed into a singular one. 
- Various sites had their commentary set to also be nav enabled, reverted this change.

Ahab's White Whale: Apple Geeks, it doesn't deserve to be, but it's doing it's darnest to be it.

Features and Big Changes:
- Full screen button. Primarily useful for Mobile users, but who knows what desktop users could use it for!  https://youtu.be/TKl02m4lkJc?t=88
    > Made it so that it'll detect when fullscreen is exited, regardless of method of doing so.
- (Mobile only) A Disable/Enable This Script button at the bottom of the page. Greasemonkey on mobile already does this, as it inserts menu commands as a button. This makes it universal regardless if it's greasemonkey or tampermonkey. As well as not a tiny button for mobile users.
- Script's buttons have a default style
   > First and Last links are now buttons. Lightblue for first, Light red for last
- Various Colors (buttons, settings menu) fixed to not be affected by OS's color scheme
- Several more color and styling fixes
- Script now does not load on image files (png, JPG, gif, etc). This was mainly an issue on mobile from what I've seen.
- Non-Dynamically Colored Buttons now get darker when pressed.
- The Dimming background now overextends width and height to fix an issue on mobile when it came to scrolling
- Pages Dropbox now have a max size of 100% (not all browsers have been updated to allow min()/max() yet, otherwise it would be 500px or 100%)
- Settings window has had various updates to give a more uniform size and feel across each page. 
    > By also forcing elements within the settings window, it helps prevent a site from overriding it and making it a mess.
    > As well as work to help to prevent drifting that occurred on some sites.
- wcr_ext_navi_ctrls: a function that can be called in js: for automatically turning the extra section into a navigable part.
    > ex `js:	 wcr_ext_navi_ctrls`

- All includes have been future proofed by switching from HTTP and HTTPS to HTTP*. Should prevent the script from breaking if a site switches between them.

Sites fixed due to switching to "http*" (checked back in october/november 2019)
biggercheese.com/
bittersweetcandybowl.com/
bringbackroomies.com/
buckocomic.com/
https://www.c.urvy.org/
collectedcurios.com/
doctorcatmd.com/
dynasty-scans.com/
garfieldminusgarfield.net/
giantitp.com/
gpf-comics.com/
little-gamers.com/
meekcomic.com/
mrlovenstein.com/
pebbleversion.com/
pointguardian.com/
project-apollo.net/mos/
satwcomic.com/
scarygoround.com/
sevensmith.net/chirault/ - Front page broken
supercrash.net/
vgcats.com/
webtoons.com/ - Slightly broken, pages don't show up till you scroll down a little (fixed? I didn't do anything, but it hasn't happened in a while)

-Post note-
 Ahab's White Whale: Apple Geeks, it doesn't deserve to be, but it's doing it's darnest to be it.
I tried, many, many, many times to get it to work with it to no avail. It's... it's weird. Maybe one day I'll try again, but if someone wants to give it ago, go ahead. It's like it ignores the capture rules and instead wants to grab the banner image as "Back", despite the xpath being completely different.

There are also other ideas I do want to implement at some point, but I need to land on stable ground, script wise, before progressing on those fronts. Sorry it took nearly a year to get this going, but, well... those changes. And procrastination.

### CODE REVIEW

**Comment by SoraHjort** [comment]
One of these days I'll properly learn and deploy pulls the first time. Maybe. 
That should fix any of the odd commits/compares.

And it looks like the silliness/woops I mentioned carried over onto the main branch, for some reason. Basically I accidentally created an "environment" and couldn't quite figure out how to remove it. Hence the parts where it says "set theme" and "delete _config.yml"

I'll contemplate if I should just do a "merge" from my fixes and additions branch with pull-requester so it has the full commit history. Since what I was trying to avoid happened anyway.

**Comment by anka-213** [comment]
You can look at how to do "interactive rebase" `rebase -i` (followed by a `push -f`) if you want to fix it. But it is also fine to leave it as it is.

**Comment by SoraHjort** [comment]
Too late, fixed it :P
I do apologize though, I am so not used to these systems. I can manage with personal edits and what not. But pulls are still foreign to me. (Also that environment mixup threw me through a loop trying to undo)

**Comment by anka-213** [comment]
No problems :)

**Comment by SoraHjort** [comment]
So finally working on this again (in a branch which I'll merge into this pull request)

The problem I'm having right now is that the english translation... made a bit of a mess of things.

Not on their part, mind you. It's gone and created a lot of conflicts that I have to go in manually.

I thought I could just use github's desktop app to rebase and what not... buuuuuuuut! It's not great. It doesn't even use the complete history of changes I made in the branch I'm working in, only up to July of last year, for some reason.

So, what I'm basically going to have to do is manually replace all the translation stuff into my fork. 

Plus it'd probably be a lot quicker than having to go through the 100+ changes it wants me to check every, single, time. It wanted to open every single conflict in my editor. A lot of them were going from four spaces to tab spaces. Though some of them are related to functions and new sites.

When I'll get this done, I do not know. I'll basically have a week free from everything, because I just moved, the ISP tech is swamped and can't setup the lines till next week at the earliest. And with the power of not-hotspotting with a smartphone, the only thing I can really do with my internet connection is browse the web, watch videos, and play games I've already downloaded.

So yeah, got free time for that now, minus some more moving related stuff.

But yeah, got a bunch of stuff to add to this pull, fixes, visual cohesion stuff, new sites, and what not in the works. Trying to keep myself from adding more stuff till I get the translation stuff pulled in.

Again, sorry it's taken so long to get around to doing this.

**Comment by SoraHjort** [comment]
Managed to get the translation inserted sooner (thanks to GitKraken. Found it earlier, not long after my last comment, it's been nice so far in it's small test). Still had to do some manual stuff. Saved me probably a few days of eye sore work.

Updated first comment to include the changes made. Not too much changed, some sites added, some sites tweaked, unified navigation buttons, some mobile related fixes (such as dimmed mode having issues while scrolling). 

Also resolved the github file conflicts that would come up, of course.

Things should be good to go for a merge in if deemed so. 

For the most part I've been using my own dev branch for the past year or so, so I can basically say that the major changes have been working just fine. And I haven't made much major additions/changes during that time due being dragged from one corner of the USA, to basically it's opposite, for a year.

It should be a while before I submit anything new to be pulled into the project. Such as more sites, as well as some feature tweaks. So there are things I still want to do, but that can wait for a later pull.

edit: also fixed this in the process https://github.com/anka-213/webcomic_reader/issues/158

**Comment by anka-213** [comment]
I’ll go ahead and merge this now so more people can get access to all your fixes
